client-side/src/components/profileComponents/profilePageComponent.jsx

```diff
@@ -0,0 +1,231 @@
+      status: website.status,
+      limitedMinutes: website.status === 'limit' ? website.limitedMinutes : '-',
+    });
+  };
+
+  const handleSave = async (id) => {
+    if (!selectedProfile || !editRowId || !editedRows) return;
+
+    const websiteAfterUpdate = {
+      websiteId: {
+        _id: id,
+        name: editedRows.name,
+        url: editedRows.url,
+      },
+      status: selectedProfile.listWebsites.find(website => website.websiteId._id === id).status,
+      limitedMinutes: selectedProfile.listWebsites.find(website => website.websiteId._id === id).status === 'limit' ? editedRows.limitedMinutes : selectedProfile.listWebsites.find(website => website.websiteId._id === id).limitedMinutes,
+    };
+
+    const updatedWebsites = selectedProfile.listWebsites.map(website =>
+      website.websiteId._id === id ? websiteAfterUpdate : website
+    );
+
+    const profileToUpdate = {
+      ...selectedProfile,
+      listWebsites: updatedWebsites
+    };
+
+    try {
+      await updateWebsite(id, { name: editedRows.name, url: editedRows.url });
+      await updateProfileApi(selectedProfile._id, profileToUpdate);
+      dispatch(updateProfile(profileToUpdate));
+      console.log('2', profiles)
+      setSelectedProfile(profileToUpdate);
+      setEditRowId(null);
+      setEditedRows(null);
+    } catch (err) {
+      console.error('Error saving profile:', err);
+    }
+  };
+
+  const handleCancel = () => {
+    setEditRowId(null);
+    setEditedRows(null);
+  };
+
+  const handleFieldChange = (e, id) => {
+    const { value, name } = e.target;
+    if (name === 'status' && editRowId === id) {
+      return;
+    }
+    if (name === 'limitedMinutes' && selectedProfile.listWebsites.find(website => website.websiteId._id === id).status !== 'limit') {
+      return;
+    }
+    setEditedRows({
+      ...editedRows,
+      [name]: value
+    });
+  };
+
+  const handleProfileUpdated = async (updatedProfile) => {
+    dispatch(updateProfile(updatedProfile));
+    navigate(0);
+    await fetchProfiles();
+    setSelectedProfile(updatedProfile);
+  };
+
+  const actions = [
+    { func: handleDelete, icon: DeleteIcon, label: 'delete', condition: (id) => id !== editRowId },
+    { func: handleEdit, icon: EditIcon, label: 'edit', condition: (id) => id !== editRowId },
+    { func: handleSave, icon: SaveIcon, label: 'save', condition: (id) => id === editRowId },
+    { func: handleCancel, icon: CancelIcon, label: 'cancel', condition: (id) => id === editRowId },
+  ];
+
+  const generateTableData = (profile) => {
+    if (!profile || !profile.listWebsites || profile.listWebsites.length === 0) {
+      return { headers: [], rows: [] };
+    }
+    const rows = profile.listWebsites
+      .filter(website => website && website.websiteId)
+      .map(website => ({
+        id: website.websiteId ? website.websiteId._id : '',
+        name: (editRowId === website.websiteId._id && editedRows) ? editedRows.name : website.websiteId.name,
```

Comment:
save editRowId === website.websiteId._id && editedRows into a variable and use it